pkg/pod/pod_test.go

```diff
@@ -76,3 +79,237 @@ func TestWithCustomResourcesRequests(t *testing.T) {
+		},
+		{
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			expectedErrMsg:   "context deadline exceeded",
+			pod:              generateTestPod("test1", "ns1", corev1.PodFailed, corev1.PodScheduled, false),
+		},
+	}
+
+	for _, testCase := range testCases {
+		var runtimeObjects []runtime.Object
+		runtimeObjects = append(runtimeObjects, testCase.pod)
+		testBuilder, err := buildPodTestBuilderWithFakeObjects(runtimeObjects, testCase.pod.Name, testCase.pod.Namespace)
+		assert.Nil(t, err)
+
+		err = testBuilder.WaitUntilHealthy(2*time.Second, testCase.includeSucceeded, testCase.checkReadiness,
+			testCase.ignoreFailedPods)
+
+		assert.Equal(t, testCase.expectedErrMsg, getErrorString(err))
+	}
+}
+
+func TestWaitForAllPodsInNamespacesHealthy(t *testing.T) {
+	testCases := []struct {
+		namespaces       []string
+		includeSucceeded bool
+		checkReadiness   bool
+		ignoreFailedPods bool
+		ignoreNamespaces []string
+		expectedErrMsg   string
+	}{
+		{
+			namespaces:       []string{"ns1"},
+			includeSucceeded: true,
+			checkReadiness:   false,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "",
+		},
+		{
+			namespaces:       []string{"ns1"},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "",
+		},
+		{
+			namespaces:       []string{"ns2"},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "context deadline exceeded",
+		},
+		{
+			namespaces:       []string{""},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{},
+			expectedErrMsg:   "context deadline exceeded",
+		},
+		{
+			namespaces:       []string{""},
+			includeSucceeded: true,
+			checkReadiness:   true,
+			ignoreFailedPods: true,
+			ignoreNamespaces: []string{"ns2"},
+			expectedErrMsg:   "context deadline exceeded",
+		},
+	}
+
+	var runtimeObjects []runtime.Object
+	runtimeObjects = append(runtimeObjects, generateTestPod("test1", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test2", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test3", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test4", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test5", "ns1", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test1", "ns2", corev1.PodRunning, corev1.PodReady, false))
+	runtimeObjects = append(runtimeObjects, generateTestPod("test2", "ns2", corev1.PodRunning, corev1.PodInitialized,
+		false))
+	fakeClient := k8sfake.NewSimpleClientset(runtimeObjects...)
```

Comment:
Instead of using the k8sfake directly, there's a function in the clients package to get the `clients.Settings` with the runtimeObjects already. [Here's an example in the CGU package.](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/cgu/cgu_test.go#L96-L98) This also allows you to clean up lines 307-309

Reply:
fixed